src/main/java/com/labsynch/labseer/chemclasses/rdkit/ChemStructureServiceRDKitImpl.java

```diff
@@ -0,0 +1,482 @@
+		} 
+		return output;
+	}
+
+	@Override
+	public MolConvertOutputDTO cleanStructure(String structure, int dim, String opts)
+			throws IOException, CmpdRegMolFormatException {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public String hydrogenizeMol(String structure, String inputFormat, String method)
+			throws IOException, CmpdRegMolFormatException {
+		RWMol mol = RWMol.MolFromMolBlock(structure);
+		if (method.equalsIgnoreCase("HYDROGENIZE")){
+			RDKFuncs.addHs(mol);		
+		} else {
+			RDKFuncs.removeHs(mol);
+		}
+		return mol.MolToMolBlock();
+	}
+
+	@Override
+	public String getCipStereo(String structure) throws IOException, CmpdRegMolFormatException {
+		//Jchem implementation returns something like
+		//e.g. [TH: 1{0,3,2} S]
+		//Indigo implementaiton returns the mol file after running some Indigo function
+		//No implementing this function because its not used anywhere that I can find
+		//Perhaps it was used as a utility function at somepoint
+		return "NOT IMPLEMENTED";
+	}
+
+	@Override
+	public StrippedSaltDTO stripSalts(CmpdRegMolecule inputStructure) throws CmpdRegMolFormatException {
+		CmpdRegMoleculeRDKitImpl molWrapper = (CmpdRegMoleculeRDKitImpl) inputStructure;
+		CmpdRegMoleculeRDKitImpl mol = molWrapper.clone();
+		RWMol clone = mol.molecule;
+		List<CmpdRegMoleculeRDKitImpl> allFrags = new ArrayList<CmpdRegMoleculeRDKitImpl>();
+	    ROMol_Vect frags = RDKFuncs.getMolFrags(clone);
+		for (int i = 0; i < frags.size(); i++) {
+			RWMol frag = (RWMol) frags.get(i);
+			CmpdRegMoleculeRDKitImpl fragWrapper = new CmpdRegMoleculeRDKitImpl(frag);
+			allFrags.add(fragWrapper);
+		}
+	
+		Map<Salt, Integer> saltCounts = new HashMap<Salt, Integer>();
+		Set<CmpdRegMoleculeRDKitImpl> unidentifiedFragments = new HashSet<CmpdRegMoleculeRDKitImpl>();
+		for (CmpdRegMoleculeRDKitImpl fragment : allFrags){
+			int[] cdIdMatches = searchMolStructures(fragment.getMolStructure(), "Salt_Structure", "DUPLICATE_TAUTOMER");
+			if (cdIdMatches.length>0){
+				Salt foundSalt = Salt.findSaltsByCdId(cdIdMatches[0]).getSingleResult();
+				if (saltCounts.containsKey(foundSalt)) saltCounts.put(foundSalt, saltCounts.get(foundSalt)+1);
+				else saltCounts.put(foundSalt, 1);
+			}else{
+				unidentifiedFragments.add(fragment);
+			}
+		}
+		StrippedSaltDTO resultDTO = new StrippedSaltDTO();
+		resultDTO.setSaltCounts(saltCounts);
+		resultDTO.setUnidentifiedFragments(unidentifiedFragments);
+		logger.debug("Identified stripped salts:");
+		for (Salt salt : saltCounts.keySet()){
+			logger.debug("Salt Abbrev: "+salt.getAbbrev());
+			logger.debug("Salt Count: "+ saltCounts.get(salt));
+		}
+		return resultDTO;
+	}
+
+	@Override
+	public String standardizeStructure(String molfile) throws CmpdRegMolFormatException, IOException {
+		// service to standardize input structure
+		// return standardized structre
+		// error conditions? 
+		// throw or catch errors
+		// create Standardizer based on a XML configuration file
```

Comment:
nit: Update these comments. There is no XML file to speak of, and the comments above seem like random scratch / notes.